File: nodecore-miners-pop/src/test/java/nodecore/miners/pop/rules/conditions/Round2ConditionTests.java
// VeriBlock PoP Miner
// Copyright 2017-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.miners.pop.rules.conditions;

import nodecore.miners.pop.Configuration;
import org.junit.Assert;
import org.junit.Test;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class Round2ConditionTests {

    @Test
    public void isActiveWhenConfigurationFalse() {
        Configuration config = mock(Configuration.class);
        when(config.getBoolean("auto.mine.round2")).thenReturn(false);

        Round2Condition condition = new Round2Condition();
        Assert.assertFalse(condition.isActive(config));
    }

    @Test
    public void isActiveWhenConfigurationTrue() {
        Configuration config = mock(Configuration.class);
        when(config.getBoolean("auto.mine.round2")).thenReturn(true);

        Round2Condition condition = new Round2Condition();
        Assert.assertTrue(condition.isActive(config));
    }

    @Test
    public void evaluateWhenHeightNull() {
        Round2Condition condition = new Round2Condition();
        Assert.assertFalse(condition.evaluate(null));
    }

    @Test
    public void evaluateWhenHeightIsNotRound2() {
        Round2Condition condition = new Round2Condition();
        for (int i = 1; i <= 19; i++) {
            if (i % 3 != 2) {
                Assert.assertFalse(condition.evaluate(10000 + i));
            }
        }
    }

    @Test
    public void evaluateWhenHeightIsRound2() {
        Round2Condition condition = new Round2Condition();
        for (int i = 1; i <= 19; i++) {
            if (i % 3 == 2) {
                Assert.assertTrue(condition.evaluate(13240 + i));
            }
        }
    }
}

File: nodecore-miners-pop/src/test/java/nodecore/miners/pop/automine/conditions/Round2ConditionTests.kt
// VeriBlock PoP Miner
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.miners.pop.automine.conditions

import io.kotlintest.shouldBe
import nodecore.miners.pop.AutoMineConfig
import nodecore.miners.pop.automine.round2Condition
import org.junit.Test

class Round2ConditionTests {
    private val config = AutoMineConfig(round2 = true)

    @Test
    fun evaluateWhenHeightIsNotRound2() {
        for (i in 1..19) {
            if (i % 3 != 2) {
                round2Condition(config, 10000 + i) shouldBe false
            }
        }
    }

    @Test
    fun evaluateWhenHeightIsRound2() {
        for (i in 1..19) {
            if (i % 3 == 2) {
                round2Condition(config, 13240 + i) shouldBe true
            }
        }
    }
}
